osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -305,12 +327,118 @@ private double computeFlashlightValue(ScoreInfo score, OsuDifficultyAttributes a
             return flashlightValue;
         }
 
+        /// <summary>
+        /// Estimates player's deviation on speed notes using <see cref="calculateDeviation"/>, assuming worst-case.
+        /// Treats all speed notes as hit circles.
+        /// </summary>
+        private double? calculateSpeedDeviation(OsuDifficultyAttributes attributes)
+        {
+            if (totalSuccessfulHits == 0)
+                return null;
+
+            // Calculate accuracy assuming the worst case scenario
+            double speedNoteCount = attributes.SpeedNoteCount;
+            speedNoteCount += (totalHits - attributes.SpeedNoteCount) * 0.1;
```

Comment:
Do we need to add the existing speed note count? If so, what's your motivation? Removing this line seems to make minimal difference to PP and makes more sense to me.

Reply:
This line is responsible for unnerfing scores like this https://osu.ppy.sh/scores/2320056008 with almost 0 negative side-effects

Reply:
Is that all there is to it? I'd prefer to have a more solid reasoning for that to exist

Reply:
It makes estimation less worst-case and therefore more reasonable
In speed pp you already you're using `(accuracy + relevantAccuracy) / 2.0` what is essentially `(attributes.SpeedNoteCount + totalHits) / 2`, what is more lenient than this, where only 10% of the normal notes are added